### ISSUE TITLE
Is there a way to figure out why flycheck can not use `lsp` checker?

### ISSUE THREAD

**Issue by zoechi** [flycheck/flycheck#1693]
This template is for **user support**.  If you are reporting an actual Flycheck bug, please use the relevant issue template instead.

## Checklist
- [x] I have searched existing issues and StackOverflow for solutions.
- [x] I have read the [Flycheck manual][].
- [x] I have read the [Troubleshooting Guide][].
- [x] I have checked my issue has not been reported yet.

## Problem description
A clear and concise description of the problem you encounter.

I use flycheck with lsp-mode with various languages successfully already.
Now I tried to make Dart-support work. LSP works, but I can't make flycheck work with dart-mode.
I tried to activate using `M-x` `flycheck-select-checker` `lsp` but I only get

```
Syntax checker in buffer test.dart in dart-mode:

  lsp
    - major mode: `dart-mode' not supported
    - may enable: yes
    - predicate:  t

Flycheck cannot use this syntax checker for this buffer.

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled
checkers.

--------------------

Flycheck version: 32snapshot (package: 20200318.319)
Emacs version:    28.0.50
System:           x86_64-pc-linux-gnu
Window system:    x
```

## Steps to reproduce
1. Open file '...'
2. Do '....'
3. See error

## System configuration
```
Syntax checkers for buffer ably.dart in dart-mode:

No checker to run in this buffer.

No checkers are available for this buffer.


The following syntax checkers are not registered:

  - lsp
  - rustic-clippy

Try adding these syntax checkers to `flycheck-checkers'.  Flycheck
Mode is enabled.  Use C-u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 32snapshot (package: 20200318.319)
Emacs version:    28.0.50
System:           x86_64-pc-linux-gnu
Window system:    x
```

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

## Additional notes
Also reported https://github.com/bradyt/dart-mode/issues/109

[Flycheck manual]:  https://www.flycheck.org/en/latest/user/troubleshooting.html
[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

**Comment by cpitclaudel** [comment]
The line that says "- major mode: `dart-mode' not supported" means that the checker's list of modes does not include `dart-mode`.  You can fix that using `flycheck-add-mode`, I think

**Comment by zoechi** [comment]
@cpitclaudel thanks that did it \o/
Why did I not need to do this for other languages?
Why did it work for https://github.com/bradyt/dart-mode/issues/109 but not for me?
Can and should this experience be improved somehow so that it can "just work" in the future like with other languages?

**Comment by zoechi** [comment]
In case someone else stumbles upon this.
Below code made it work for me. 
Suggestions for improvement are welcome.

```list
(use-package
  dart-mode
  :ensure t
  :config (setq dart-format-on-save t)
  :hook ((dart-mode . (lambda ()
                        (flycheck-add-mode 'lsp 'dart-mode)
                        (flycheck-select-checker 'lsp)))))
```

**Comment by cpitclaudel** [comment]
> Why did I not need to do this for other languages?

Normally `lsp-flycheck-enable` calls `lsp-flycheck-add-mode`, which does this.

> Why did it work for bradyt/dart-mode#109 but not for me?

Not sure, you'd have to compare your config to theirs

> Can and should this experience be improved somehow so that it can "just work" in the future like with other languages?

I think it should already work; can you check whether the config in the other issue works for you too?
